crl/x509/crl.go

```diff
@@ -254,16 +274,52 @@ func CreateRevocationList(rand io.Reader, template *RevocationList, issuer *x509
 		return nil, errors.New("x509: template contains nil Number field")
 	}
 
+	numBytes := template.Number.Bytes()
+	if len(numBytes) > 20 || numBytes[0]&0x80 != 0 && len(numBytes) > 19 {
```

Comment:
Nit, some parentheses would make this expression easier to parse.

```suggestion
	if len(numBytes) > 20 || (numBytes[0]&0x80 != 0 && len(numBytes) > 19) {
```